api/app/Generators/DocGenerator.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace App\Generators;
+
+use Illuminate\Support\Facades\Storage;
+use PhpOffice\PhpWord\Element;
+use PhpOffice\PhpWord\IOFactory;
+use PhpOffice\PhpWord\PhpWord;
+
+abstract class DocGenerator
+{
+    protected PhpWord $doc;
+
+    protected array $strong;
+
+    public function __construct()
+    {
+        $this->doc = new PhpWord();
+
+        $this->doc->addTitleStyle(1, ['size' => 22, 'bold' => true], ['spaceBefore' => 240, 'spaceAfter' => 120]);
+        $this->doc->addTitleStyle(2, ['size' => 18, 'bold' => true], ['spaceBefore' => 240, 'spaceAfter' => 120]);
+        $this->doc->addTitleStyle(3, ['size' => 15, 'bold' => true], ['spaceBefore' => 240, 'spaceAfter' => 120]);
+        $this->doc->addTitleStyle(4, ['size' => 13, 'bold' => true], ['spaceBefore' => 240, 'spaceAfter' => 120]);
+        $this->doc->addTitleStyle(4, ['size' => 12, 'bold' => true], ['spaceBefore' => 240, 'spaceAfter' => 120]);
+
+        $this->strong = ['bold' => true];
+    }
+
+    abstract public function generate();
+
+    public function write(string $fileName)
+    {
+        /** @var \Illuminate\Filesystem\FilesystemManager */
+        $disk = Storage::disk('user_generated');
+
+        $path = $disk->path($fileName);
+
+        $writer = IOFactory::createWriter($this->doc);
+
+        return $writer->save($path);
+    }
+
+    protected function addLabelText(Element\Section $section, string $label, string $text)
+    {
+        $run = $section->addTextRun();
+        $run->addText($label.': ', $this->strong);
+        $run->addText($text);
+    }
+
+    protected function santizeEnum(string $enum): string
```

Comment:
```suggestion
    protected function sanitizeEnum(string $enum): string
```

Reply:
oof, I'm blaming this one on copilot! It auto completed this definition for me because i had a similar one I was playing with and copilot propagated a previous mistake :confounded: 

Anyway, should be fixed in [fix typo in function name](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9890/commits/fa995facf3bb90b4eec569d5a526cfb591c6f7db)